File: Programming Basics with Python/Вложени цикли/train_the_trainers.py

n = int(input())
presentation_name = input()

count_grades = 0
sum_all_grades = 0
while presentation_name != "Finish":
    current_sum_grades = 0
    for i in range(1, n+1):
        grade = float(input())
        count_grades += 1
        sum_all_grades += grade
        current_sum_grades += grade

    avg_of_current = current_sum_grades / n
    print(f"{presentation_name} - {avg_of_current:.2f}.")

    presentation_name = input()

avg_result = sum_all_grades / count_grades
print(f"Student's final assessment is {avg_result:.2f}.")

File: Programming Basics with Java/Nested Loops/TrainTheTrainers.java
import java.util.Scanner;

public class TrainTheTrainers {
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int juryCount = Integer.parseInt(input.nextLine());
        String presentationName = input.nextLine();

        double totalAverageGrade = 0;
        int presentationCount = 0;

        while (!presentationName.equals("Finish")) 
        {
            double sumGrades = 0;

            for (int i = 0; i < juryCount; i++) 
            {
                double grade = Double.parseDouble(input.nextLine());
                sumGrades += grade;
            }

            double averageGrade = sumGrades / juryCount;
            totalAverageGrade += averageGrade;
            presentationCount++;

            System.out.printf("%s - %.2f.%n", presentationName, averageGrade);
            presentationName = input.nextLine();
        }

        double finalAssessment = totalAverageGrade / presentationCount;
        System.out.printf("Student's final assessment is %.2f.", finalAssessment);
    }
}
